honeybee_radiance/lightsource/sky/certainirradiance.py

```diff
@@ -0,0 +1,160 @@
+"""Generate sky with certain irradiance."""
+from __future__ import division
+from ._skybase import Sky
+from .hemisphere import Hemisphere
+from honeybee_radiance.lightsource.ground import Ground
+import honeybee_radiance.typing as typing
+import ladybug.futil as futil
+
+
+class CertainIrradiance(Sky):
+    """sky with certain irradiance.
+
+    The output of CertainIrradiance sky is similar to using command below:
+    `gensky -c -B desired_irradiance `
+
+    You can also generate the sky with certain illuminance using `from_illuminance`
+    classmethod. The method converts the illuminance value to irradiance by dividing it
+    by 179.0.
+
+    `gensky -c -B [desired_illuminance / 179.0]`
+
+    It also includes ground glow source. Ground reflectance is set to %20 by default
+    which is gensky's default value. Use `ground_reflectance` property to adjust this
+    value.
+    """
+    __slots__ = (
+        '_irradiance', '_ground', '_ground_reflectance',
+        '_ground_hemisphere', '_sky_hemisphere'
+    )
+
+    def __init__(self, irradiance=558.659, ground_reflectance=0.2):
+        """Create sky with certain irradiance.
+
+        Default value is set to 558.659 which corresponds to a sky with 100,000 lux
+        horizontal illuminance.
+
+        Args:
+            irradiance: Desired horizontal diffuse irradiance value in watts/meter2
+                (Default: 558.659).
+            ground_reflectance: Average ground reflectance (Default: 0.2).
+        """
+        Sky.__init__(self)
+        self.irradiance = irradiance
+        self.ground_reflectance = ground_reflectance
+        self._ground_hemisphere = Ground()
+        self._sky_hemisphere = Hemisphere()
+
+
+    @classmethod
+    def from_illuminance(cls, illuminance=100000, ground_reflectance=0.2):
+        """Create sky with certain illuminance.
+
+        Args:
+            illuminance: Desired horizontal illuminance value in lux (Default: 100000).
+            ground_reflectance: Average ground reflectance (Default: 0.2).
+        """
+        return cls(illuminance / 179.0, ground_reflectance)
+
+    @property
+    def irradiance(self):
+        """Sky irradiance value."""
+        return self._irradiance
+
+    @irradiance.setter
+    def irradiance(self, irradiance):
+        self._irradiance = typing.float_positive(irradiance) \
+            if irradiance is not None else 558.659
+
+    @property
+    def illuminance(self):
+        """Sky illuminance value."""
+        return round(self._irradiance * 179.0, 2)
```

Comment:
This 179 factor is confusing (I can see why Greg gets a lot of question about it).  If `illuminance` is going to be a property here, I think you should explain where it comes from in the docstring.  Or perhaps you can tell people, if they actually need to estimate illuminance values from irradince, they should not use this property but should be using [Perez's luminous efficacy model that is implemented in ladybug core](https://github.com/ladybug-tools/ladybug/blob/master/ladybug/skymodel.py#L236-L352)

Reply:
Yes. This is specific to Radiance. You can read more about this on the Radiance forum including [this one](https://discourse.radiance-online.org/t/coefficient-179/547) and [this one](https://discourse.radiance-online.org/t/luminous-efficacy/1400).

> Or perhaps you can tell people, if they actually need to estimate illuminance values from irradince, they should not use this property but should be using Perez's luminous efficacy model that is implemented in ladybug core

Not sure if we want to make it complicated for people. I don't know why someone would use this method which is for a cloudy sky with a certain illuminance level to estimate illuminance value from irradiance at the first point.